packages/convertkit/src/hooks/use-convertkit-form.tsx

```diff
@@ -0,0 +1,50 @@
+import {useFormik} from 'formik'
+import * as Yup from 'yup'
+import axios from 'axios'
+import {ConvertkitSubscriber} from '../types'
+import {
+  CONVERTKIT_SIGNUP_FORM,
+  CONVERTKIT_SUBSCRIBE_API_URL,
+} from '@skillrecordings/config'
+
+export function useConvertkitForm({
+  submitUrl = CONVERTKIT_SUBSCRIBE_API_URL,
+  formId = (CONVERTKIT_SIGNUP_FORM || 0) as number,
+  onSuccess,
+  onError,
+}: {
+  submitUrl?: string
+  formId?: number
+  onSuccess: (subscriber: ConvertkitSubscriber) => void
+  onError: (error?: any) => void
+}) {
+  const {isSubmitting, status, handleChange, handleSubmit} = useFormik({
+    initialStatus: '',
+    initialValues: {
+      email: '',
+      first_name: '',
+    },
+    validationSchema: Yup.object().shape({
+      email: Yup.string().email('Invalid email address').required('Required'),
+      first_name: Yup.string(),
+    }),
+    validateOnChange: false,
+    enableReinitialize: true,
+    onSubmit: async ({email, first_name}, {setStatus}) => {
+      return axios
```

Comment:
returning this here is important so that the promise will resolve. Eliminated the need to independently track `isSubmitted`